### PR TITLE
Fix for the crash in pre_transform_target_entry

### PR DESCRIPTION
**Pull request by riyajain39**
### Description

This pull request fix the crash is `pre_transform_target_entry` for queries that involves DELETE ... OUTPUT with JOIN statement. 

**Issue:**
For the delete queries with join statement, somehow the `res->location` is not processed correctly. This causes server to crash as the value of `last_dot` pointer has never been computed. I implemented a check which first computed whether the `res->location` is correct or not and then further proceed with the code.

### Issues Resolved

BABEL-4484

Signed-off-by: Riya Jain [riyaajn@amazon.com](mailto:riyaajn@amazon.com)

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/hooks.c [hunk 1]
```diff
@@ -1667,13 +1668,14 @@ pre_transform_target_entry(ResTarget *res, ParseState *pstate,
 			 * Case 3: Handle the case when column name is delimited with sqb. When number of sqb
 			 * are zero, it means we are out of sqb.
 			 */
-			if(list_length(cref->fields) > 1 &&
-				IsA(llast(cref->fields), String))
+			if(res->location != 0 && (list_length(cref->fields) > 1 &&
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/hooks.c]: This should be exclusive condition, hence use else if

File: contrib/babelfishpg_tsql/src/hooks.c [hunk 2]
```diff
@@ -1643,10 +1643,11 @@ pre_transform_target_entry(ResTarget *res, ParseState *pstate,
 			/*
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/hooks.c:1620]: Please add few test cases from jira

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/src/hooks.c (hunk 2 above)
Comment (riyajain39) [reply contrib/babelfishpg_tsql/src/hooks.c:1620]: Resolved.

File: contrib/babelfishpg_tsql/src/hooks.c [hunk 3]
```diff
@@ -1721,7 +1723,7 @@ pre_transform_target_entry(ResTarget *res, ParseState *pstate,
 			int actual_alias_len = 0;
 
 			/* To handle queries like SELECT ((<column_name>)) from <table_name> */
-			while(*colname_start == '(' || scanner_isspace(*colname_start))
+			while(*colname_start != '\0' && (*colname_start == '(' || scanner_isspace(*colname_start)))
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/hooks.c:1726]: colname_start can still be uninitialised. Initialised it to NULL on line 1609

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/src/hooks.c (hunk 3 above)
Comment (riyajain39) [reply contrib/babelfishpg_tsql/src/hooks.c:1726]: Resolved

**Review by Deepesh125 — APPROVED** [review]
